ow_dynamic_terrain/src/TerrainModifier.cpp

```diff
@@ -243,29 +243,31 @@ bool TerrainModifier::applyImageToHeightmap(Heightmap* heightmap, const cv::Poin
     return false;
   }
 
-  auto heightmap_size = static_cast<int>(terrain->getSize());
-  auto left = max(center.x - image.cols / 2, 0);
-  auto top = max(center.y - image.rows / 2, 0);
-  auto right = min(center.x - image.cols / 2 + image.cols - 1, heightmap_size);
-  auto bottom = min(center.y - image.rows / 2 + image.rows - 1, heightmap_size);
+  int heightmap_size = static_cast<int>(terrain->getSize());
+  int left = center.x - image.cols / 2;
+  int top = center.y - image.rows / 2;
+  int right = left + image.cols - 1;
+  int bottom = top + image.rows - 1;
```

Comment:
This is the first time I've seen changing `auto` to a specific type fix a problem, though I can see how it would in this case.

There are many uses of `auto` earlier in the file.  Should some of those be converted too?

Reply:
To be fair, changing auto to int didn't help but it was one of the things I tried. The changes to the math are what fixed the problem. I left the changes from auto to int because int is easier to type and makes the code easier to understand. I agree that auto is overused in this file.